### ISSUE TITLE
Icecast-kh instead of icecast

### ISSUE THREAD

**Issue by amavarick** [LibreTime/libretime#406]
Would you consider switching to icecast-kh instead of icecast?  Specifically there are improvements to secure connections that icecast-kh has that icecast doesn't.  Namely I can use both encrypted and unencrypted connections on the same ports and I can specify the private key rather than chaining all of the keys into one file.  It also seems to be a more actively updated product.

**SSL updates**
build fix where no SSL is available
allow for DH/ECDH
allow for reading the cert before dropping privs, but keep certificate if reload is fails.
crash fix if no SSL certificate defined.
small fix for internal openssl threading.
autodetect SSL connections on incoming sockets. No need for <ssl> in listen-socket now but is still there for compatability.
add <ssl-private-key> in <paths> to allow for combined PEM or for separate SSL key/certificate files.
select https/http URL in autogenerated m3u based on incoming request.

**Comment by Robbt** [comment]
I have heard of Icecast-kh and it sounds like a useful and interesting project. I don't know that we'd want to replace icecast completely for our detault  with it but someone could work on documenting how to use icecast-kh instead of icecast.

It just seems like it is primarily a project of a single developer w/o a lot of documentation on the website - https://karlheyes.github.io/ whereas icecast is one of the few packages that works out of the box as is.

**Comment by yarik2720** [comment]
You can switch to icecast-kh manually, just build from source and install. Airtime and Libretime both works well with it without any limitations. Documentation can be used from official icecast website.
But there is no built packages or repos of icecast-kh so I think it can't be used by default in Libretime project

**Comment by hairmare** [comment]
We just looked into this at our station recently. We where interested in icecast-kh for being able to reload the cert without kicking all listeners. We found that most of the features in icecast-kh have already been landed in the mainline icecast version and we are waiting for the next icecast release.
Our primary use case was Lets Encrypt and we ended up buying a non Let's Encrypt cert that is valid for three years from now as a stop gap measure. We hope that there will be another icecast release that supports cert reloading before that cert expires.

LibreTime won't be switching to icecast-kh in the default install. Regular icecast is what is tried and tested on most platforms. It should be noted that lots of features that start in the kh fork eventually make it's way back into mainline icecast (AFAIK most of the SSL parts).

Depending on you OS there might also be nice packages to install the kh form from (at least Arch has some, not that it's relevant for our case). Properly done such packages should work without any hickups during the install but building them is certainly out of scope for LibreTime.
